en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+    _rates_setpoint->update(rates_sp, thrust_sp);
+  }
+
+private:
+  std::shared_ptr<px4_sdk::ManualControlInput> _manual_control_input;
+  std::shared_ptr<px4_sdk::RatesSetpointType> _rates_setpoint;
+};
+```
+
+- **[1]**: First we create a class that inherits from `px4_sdk::ModeBase`
+- **[2]**: In the constructor, we pass the mode name. This also allows to configure some other things, like replacing an FMU-internal mode.
+- **[3]**: This is where we create all objects that we want to use later on.
+  This can be RC input, setpoint type(s), or telemetry. `*this` is passed as a `Context` to each object, which associates the object with the mode.
+- **[4]**: Whenever the mode is active, this method gets called regularly (the update rate depends on the setpoint type).
+  Here is where we can do our work and generate a new setpoint.
+
+## How to use the SDK
+
+General notes on how to use the SDK:
+
+- coordinate convention: TODO
+
+The following sections describe specific functionality.
+Apart from that, any other PX4 topic can be subscribed or published directly.
+
+### Setpoint Types
+
+A mode can choose its setpoint type(s) it wants to use to control the vehicle.
+The used types also define the compatibility with different vehicle types.
+
+The following sections provide a list of commonly used setpoint types.
+You can also add your own type by adding a class that inherits from `px4_sdk::SetpointBase`, sets the configuration flags according to what the setpoint requires, and then publishes any topic containing a setpoint.
+
+#### Smooth Position
+
+TODO
+
+#### Direct Actuator Control
+
+Actuators can be directly controlled using the [px4_sdk::DirectActuatorsSetpointType](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/include/px4_sdk/control/setpoint_types/direct_actuators.hpp) setpoint type.
+Motors and servos can be set independently. Be aware that the assignment is vehicle and setup-specific.
+For example to control a quadrotor, you need to set the first 4 motors according to its [output configuration](../concept/control_allocation.md).
+
+:::note
+If you want to control an actuator that does not control the vehicle's motion, but for example a payload servo, see [below](#controlling-an-independent-actuator-servo).
+:::
+
+### Controlling an Independent Actuator/Servo
+
+If you want to control an independent actuator (a servo), follow these steps:
+
+1. [Configure the output](../payloads/#generic-actuator-control-with-mavlink)
+2. Create an instance of [px4_sdk::OffboardActuatorControls](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/include/px4_sdk/control/offboard_actuators.hpp) in the constructor of your mode
+3. Call the `set` method to control the actuator(s). This can be done independently of any active setpoints.
+
+### Telemetry
+
+Telemetry, such as local or global position estimates can be found under [px4_sdk/odometry](https://github.com/PX4/px4_sdk/tree/main/px4_sdk_cpp/include/px4_sdk/odometry).
```

Comment:
What is special about this? I mean there are other ways to get telemetry in ROS, so 
- why did you choose to create this
- can you give a list of the main telemetry items covered - reduce amount of digging
- An example might be nice "This is used by modes that need positioning to plan future movement blah blah"

Reply:
A few reasons:
- for convenience
- changing the message does not mean you have to change your code
- using them automatically sets the mode requirement flags

I expect more things to be added in near future.